openstack/swift/templates/proxy-deployment.yaml

```diff
@@ -43,23 +45,42 @@ spec:
     spec:
       {{- if $cluster.proxy_deployment_to_storage_nodes }}
       {{- include "swift_daemonset_tolerations" $ | indent 6 }}
-      nodeSelector:
-        species: {{ $.Values.species }}
       {{- end }}
       affinity:
-        podAntiAffinity:
+        nodeAffinity:
+          # Only schedule on nodes in this az
           requiredDuringSchedulingIgnoredDuringExecution:
-          - labelSelector:
-              matchExpressions:
-              - key: component
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: failure-domain.beta.kubernetes.io/zone
                 operator: In
                 values:
-                - swift-proxy-{{$cluster_id}}
-              - key: from
-                operator: In
+                - {{ $az }}
+          # Prefere non storage nodes for scheduling
+          preferredDuringSchedulingIgnoredDuringExecution:
```

Comment:
Would prefer to have this as a hard requirement, to avoid surprises.

Reply:
done